src/lib.rs

```
@@ -160,25 +197,23 @@ mod tests {
     }
 
     #[test]
-    #[cfg(not(any(target_os = "bitrig", target_os = "openbsd")))]
+    #[cfg(not(target_os = "openbsd"))]
     fn test_pledge_unsupported() {
         use super::Error;
-        assert_eq!(pledge![Stdio].unwrap_err(), Error::UnsupportedPlatform);
+        assert_eq!(pledge_promises![Stdio].unwrap_err(), Error::UnsupportedPlatform);
     }
 
     #[test]
-    #[cfg(any(target_os = "bitrig", target_os = "openbsd"))]
+    #[cfg(target_os = "openbsd")]
     fn test_pledge_supported() {
-        pledge![Stdio].unwrap();
-        assert!(pledge![Stdio, Audio].is_err());
+        pledge_promises![Stdio].unwrap();
+        assert!(pledge_promises![Stdio Audio].is_err());
     }
 
     #[test]
     #[cfg(target_os = "openbsd")]
     fn test_as_string() {
-        if pledge("stdio").is_err() {
-            panic!("pledge");
-        }
```

using unwrap here panics without swallowing details about the Err